src/errors.rs

```diff
@@ -1,68 +1,68 @@
+        // Validation error
 
-impl error::Error for Error {
-    fn description(&self) -> &str {
-        match *self {
-            Error::EncodeJSON(ref err) => err.description(),
-            Error::DecodeBase64(ref err) => err.description(),
-            Error::DecodeJSON(ref err) => err.description(),
-            Error::Utf8(ref err) => err.description(),
-            Error::InvalidToken => "Invalid Token",
-            Error::InvalidSignature => "Invalid Signature",
-            Error::WrongAlgorithmHeader => "Wrong Algorithm Header",
+        /// When a token’s `exp` claim indicates that it has expired
+        ExpiredSignature {
+            description("expired signature")
+            display("Expired Signature")
+        }
+        /// When a token’s `iss` claim does not match the expected issuer
+        InvalidIssuer {
+            description("invalid issuer")
+            display("Invalid Issuer")
+        }
+        /// When a token’s `aud` claim does not match one of the expected audience values
+        InvalidAudience {
+            description("invalid audience")
+            display("Invalid Audience")
+        }
+        /// When a token’s `aud` claim does not match one of the expected audience values
+        InvalidSubject {
+            description("invalid subject")
+            display("Invalid Subject")
+        }
+        /// When a token’s `iat` claim is in the future
+        InvalidIssuedAt {
+            description("invalid issued at")
+            display("Invalid Issued At")
+        }
+        /// When a token’s nbf claim represents a time in the future
+        ImmatureSignature {
+            description("immature signature")
+            display("Immature Signature")
         }
     }
 
-    fn cause(&self) -> Option<&error::Error> {
-        Some(match *self {
-            Error::EncodeJSON(ref err) => err as &error::Error,
-            Error::DecodeBase64(ref err) => err as &error::Error,
-            Error::DecodeJSON(ref err) => err as &error::Error,
-            Error::Utf8(ref err) => err as &error::Error,
-            ref e => e as &error::Error,
-        })
-    }
-}
-
-impl fmt::Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Error::EncodeJSON(ref err) => fmt::Display::fmt(err, f),
-            Error::DecodeBase64(ref err) => fmt::Display::fmt(err, f),
-            Error::DecodeJSON(ref err) => fmt::Display::fmt(err, f),
-            Error::Utf8(ref err) => fmt::Display::fmt(err, f),
-            Error::InvalidToken => write!(f, "{}", error::Error::description(self)),
-            Error::InvalidSignature => write!(f, "{}", error::Error::description(self)),
-            Error::WrongAlgorithmHeader => write!(f, "{}", error::Error::description(self)),
-        }
+    foreign_links {
+        Unspecified(ring::error::Unspecified) #[doc = "An error happened while signing/verifying a token with RSA"];
```

Comment:
That's the way to add comments to `foreign_links` in `error-chain`, not the prettiest